packages/plugin-core/src/telemetry.ts

```diff
@@ -3,11 +3,17 @@ import { SegmentClient } from "@dendronhq/common-server";
 import { Logger } from "./logger";
 
 /** Creates a SegmentClient for telemetry, if enabled, and listens for vscode telemetry settings to disable it when requested. */
-export function setupSegmentClient(ws: DWorkspaceV2, cachePath?: string) {
+export function setupSegmentClient({
+  ws,
+  cachePath,
+}: {
+  ws?: DWorkspaceV2;
```

Comment:
I'm not sure I completely understand - when will this not be set here?  Will we have any issues of not respecting user's disableTelemetry choice because of a race condition?

Reply:
it won't be set when the user doesn't activate dendron inside of a dendron workspace. we read the `dendron.yml` to see if they have a workspace specific override for telemetry. otherwise, we check $HOME/.dendron.telemetry file for the status